flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -346,63 +329,30 @@ private TableSchema getProducedTableSchema() {
 	}
 
 	@Override
-	public boolean isLimitPushedDown() {
-		return isLimitPushDown;
+	public void applyLimit(long limit) {
+		this.limit = limit;
 	}
 
 	@Override
-	public TableSource<RowData> applyLimit(long limit) {
-		return new HiveTableSource(
-				jobConf,
-				flinkConf,
-				tablePath,
-				catalogTable,
-				remainingPartitions,
-				hiveVersion,
-				partitionPruned,
-				projectedFields,
-				true,
-				limit);
+	public Optional<List<Map<String, String>>> listPartitions() {
+		return Optional.empty();
 	}
 
 	@Override
-	public List<Map<String, String>> getPartitions() {
-		throw new UnsupportedOperationException(
-				"Please use Catalog API to retrieve all partitions of a table");
+	public void applyPartitions(List<Map<String, String>> remainingPartitions) {
+		if (catalogTable.getPartitionKeys() != null && catalogTable.getPartitionKeys().size() != 0) {
+			this.remainingPartitions = remainingPartitions;
+		}
```

Comment:
Should we report error for the else branch ?

Reply:
I think we can